### PR TITLE
Pt object

### PR DESCRIPTION
**Pull request by azime**
Commercial modes added in StopArea Object

### CODE REVIEW

**Comment by TeXitoi** [comment]
jenkins, test this please.

File: source/type/type.cpp [hunk 1]
```diff
@@ -446,10 +446,26 @@ std::vector<idx_t> Calendar::get(Type_e type, const PT_Data & data) const{
     }
     return result;
 }
-std::vector<idx_t> StopArea::get(Type_e type, const PT_Data &) const {
+std::vector<idx_t> StopArea::get(Type_e type, const PT_Data & data) const {
     std::vector<idx_t> result;
     switch(type) {
-    case Type_e::StopPoint: return indexes(this->stop_point_list);
+    case Type_e::StopPoint: return indexes(this->stop_point_list); break;
```
Comment (TeXitoi) [inline source/type/type.cpp]: break here is useless as you return

File: source/type/type.cpp [hunk 2]
```diff
@@ -655,6 +671,20 @@ std::vector<idx_t> StopPoint::get(Type_e type, const PT_Data &) const {
...
+            }
+        }
+        break;
+    case Type_e::PhysicalMode:
+        for(const auto& jpp: journey_pattern_point_list){
+            if(jpp->journey_pattern && jpp->journey_pattern->physical_mode){
+                result.push_back(jpp->journey_pattern->physical_mode->idx);
+            }
+        }
+        break;
```
Comment (TeXitoi) [inline source/type/type.cpp:701]: You should unique the types as in StopArea::get using a std::set, no?

File: source/type/type.cpp (hunk 1 above)
Comment (azime) [inline source/type/type.cpp]: OK

**Comment by azime** [comment]
corrections

**Comment by TeXitoi** [comment]
@azime forgot to  push?

**Comment by kinnou02** [comment]
jenkins, test this please.

**Pull request merged**